src/display.rs

```diff
@@ -166,8 +165,14 @@ impl<'a> DisplayByteSlice<'a> {
 
         let pad_right = if let Some(width) = f.width() {
             let string_len = match f.precision() {
-                Some(max) if self.bytes.len() * 2 > (max + 1) / 2 => max,
-                Some(_) | None => self.bytes.len() * 2,
+                #[rustfmt::skip]
+                Some(max) => {
+                    // Odd max width which makes no sense for hex strings, honour max but use even length.
+                    let max = if max % 2 == 1 { max - 1 } else { max };
```

Comment:
Branchless: `let max = max - (max % 2);` (Maybe the compiler does it I'm not sure.)

Reply:
That suggestion is hot. Legendary.